fatiando/seismic/conv.py

```diff
@@ -0,0 +1,251 @@
+  the model in time.
+* :func:`~fatiando.seismic.conv.rickerwave`: calculates a ricker wavelet.
+
+Examples
+--------
+.. plot::
+    :include-source:
+    :context:
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from fatiando.seismic import conv
+    >>> from fatiando.vis import mpl
+    >>> # Choose some velocity depth model
+    >>> n_samples, n_traces = [600, 20]
+    >>> rock_grid = 1500.*np.ones((n_samples, n_traces))
+    >>> rock_grid[300:, :] = 2500.
+    >>> # Convert from depth to time
+    >>> [vel_l, rho_l] = conv.depth_2_time(rock_grid, dt=2.e-3)
+    >>> # Calculate the reflectivity for all the model
+    >>> rc = conv.reflectivity(vel_l,rho_l)
+    >>> # Convolve the reflectivity with a ricker wavelet
+    >>> synt = conv.convolutional_model(rc, 30., conv.rickerwave)
+    >>> # Plot the result
+    >>> fig = plt.figure(figsize=(6,5))
+    >>> _ = mpl.seismic_wiggle(synt, dt=2.e-3)
+    >>> _ = mpl.seismic_image(synt, dt=2.e-3,
+    ...                            cmap=mpl.pyplot.cm.jet, aspect='auto')
+    >>> _ = plt.ylabel('time (seconds)')
+    >>> _ = plt.title("Convolutional seismogram", fontsize=13, family='sans-serif')
+
+References
+----------
+
+Yilmaz, Oz,
+Ch.2 Deconvolution. In: YILMAZ, Oz. Seismic Data Analysis: Processing,
+Inversion, and Interpretation of Seismic Data. Tulsa: Seg, 2001. Cap. 2.
+p. 159-270. Available at: <http://dx.doi.org/10.1190/1.9781560801580.ch2>
+
+
+"""
+from __future__ import division
+import numpy as np
+from scipy import interpolate  # linear interpolation of velocity/density
+
+
+def convolutional_model(rc, f, wavelet, dt=2.e-3):
+    """
+    Calculate convolutional seismogram for a geological model
+
+    Calculate the synthetic convolutional seismogram of a geological model, Vp
+    is mandatory while density is optional. The given model in a matrix form is
+    considered a mesh of square cells.
+
+    .. warning::
+
+        Since the relative difference between the model is the important, being
+        consistent with the units chosen for the parameters is the only
+        requirement, whatever the units.
+
+    Parameters:
+
+    * rc : 2D-array
+        reflectivity values in the time domain
+    * f : float
+        Dominant frequency of the ricker wavelet
+    * wavelet : float
+        The function to consider as source in the seismic modelling.
+    * dt: float
+        Sample time of the ricker wavelet and of the resulting seismogram
+
+    Returns:
+
+    * synth_l : 2D-array
+        Resulting seismogram
+
+    """
+    w = wavelet(f, dt)
+    # convolution
+    synth_l = np.zeros(np.shape(rc))
+    for j in range(0, rc.shape[1]):
+        if np.shape(rc)[0] >= len(w):
+            synth_l[:, j] = np.convolve(rc[:, j], w, mode='same')
+        else:
+            aux = np.floor(len(w)/2.)
+            synth_l[:, j] = np.convolve(rc[:, j], w, mode='full')[aux:-aux]
+    return synth_l
+
+
+def reflectivity(model_t, rho=1.):
```

Comment:
As with `dt`, I don't think `rho`  should have a default value. Defaults are better when the user doesn't really  need to think about this, only for advanced use. This is not the case with `rho`. Forgetting to pass `rho` would lead  to wrong reflectivity.